sky/clouds/service_catalog/data_fetchers/fetch_azure.py

```diff
@@ -244,11 +244,79 @@ def get_additional_columns(row):
         axis='columns',
     )
 
+    def create_gpu_map(df):
+        # Map of Azure's machine with GPU to their corresponding memory
+        # Result is hard-coded since Azure's API to not return such info
+        # may be outdated so need to be maintained
+        gpu_map = {
```

Comment:
could we try to map instance type -> gpu name first and then calculate the resulting device memory later? this two-level approach might be cleaner.

Reply:
I agree this approach is much cleaner as it uses much less hard-coding and utilize already fetched info. I will change the script to use this approach.